docs/development/component-checklist.md

```diff
@@ -125,6 +125,12 @@ This document provides a checklist for them that you can walk through.
 
    <sub>[*] It is required because if a component deployed in the Shoot cluster does not specify a Seccomp profile and cannot run with the `RuntimeDefault` Seccomp profile, then enabling the `.spec.kubernetes.kubelet.seccompDefault` field in the Shoot spec would break the corresponding component.</sub>
 
+9. **Use strict mode when decoding data** ([example](https://github.com/gardener/gardener/blob/485c25124ea536e4610c627109017dd34d434921/pkg/provider-local/controller/worker/actuator.go#L52))
```

Comment:
I recall that it is not that easy to impose strict decoding when you already have existing resources in the system which are not compliant. It would be nice to add note/warning the backwards-compatibility problem and suggestions how to mitigate it:
- enforce strict decoding for newly created resources only
- use non-strict decoder for `oldObj` and use strict decoder for `newObj`. However, this is still backwards-incompatible. But if you use a strict decoder for `oldObj` then you cannot update the resource.